base/sparse/higherorderfns.jl

```diff
@@ -871,11 +874,11 @@ promote_containertype(::Type{AbstractSparseArray}, ::Type{Tuple}) = Array
 
 # broadcast[!] entry points for combinations of sparse arrays and other (scalar) types
 @inline function broadcast_c{N}(f, ::Type{AbstractSparseArray}, mixedargs::Vararg{Any,N})
-    parevalf, passedargstup = capturescalars(f, mixedargs)
+    parevalf, passedargstup = capturescalars(f, map(_sparsifystructured, mixedargs))
```

Comment:
Yeah, a mechanism for avoiding the need of `capturescalars` (similar to the way things work for `Array`s) is what I meant, but we can discuss that later.